web/server/api/endpoints/lineage.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import collections
+import traceback
+import typing as t
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlglot import exp, parse_one
+from sqlglot.lineage import Node, lineage
+from starlette.status import HTTP_422_UNPROCESSABLE_ENTITY
+
+from sqlmesh.core.context import Context
+from web.server.settings import get_loaded_context
+
+router = APIRouter()
+
+
+def _get_table(node: Node) -> str:
+    """Get a node's table/source"""
+    if isinstance(node.expression, exp.Table):
+        return (
+            f"{node.expression.db}.{node.expression.this}"
+            if node.expression.db
+            else str(node.expression.this)
+        )
+
+    else:
+        return node.alias
+
+
+def _process_downstream(downstream: t.List[Node]) -> t.Dict[str, t.List[str]]:
+    """Aggregate a list of downstream nodes by table/source"""
+    graph = collections.defaultdict(list)
+    for node in downstream:
+        column: exp.Column = parse_one(node.name)
+        table = _get_table(node)
+        graph[table].append(str(column.this))
```

Comment:
what's the goal of this?

Reply:
The goal is to transform the list of downstream nodes into a format the UI can use to generate a graph. Here, it's a mapping of a node's downstream tables/sources to their columns. It's a format @mykalmax and I agreed on. Happy to provide more context if needed.

Reply:
why do you do parse_one() and then str?

Reply:
Oh, I have since changed it to `column.name` based on another comment of yours in this PR, but originally, parse_one() returned an exp.Column object, which isn't JSON serializable. I cast it to str to get the column name because I didn't realize I could've used `column.name`. As for the parse_one() part, node.name could be the string, "oi.quantity", and I want to extract the column name, "quantity", from it without the table part, "oi".

Reply:
why would you do parse_one() when node.name extracts quanity from oi.quantity?

Reply:
node.name contains the table "oi" and not just "quantity".